pkg/client/unversioned/request.go

```diff
@@ -648,12 +687,70 @@ func (r *Request) Stream() (io.ReadCloser, error) {
 	}
 }
 
+type URLBackoff struct {
```

Comment:
@timothysc is in agreement.  Shall we refactor URLBackoff into utils? Or a better place?